Fix/form style

Amber-Song opened the pull request:
# Context

Link to [Notion ticket](https://www.notion.so/cloudcannon/Form-fixes-f5653be818dd47859d940f9b1646afdd?pvs=4)

# Additional information

I didn't do the "Comments should be added for form best practices" though.

# Testing (tester)

The reviewer should check on this branch:

- The code
- The component in Bookshop browser
- The site in CloudCannon (link here)

# Before merge (PR owner)

- [ ] Delete the test site in CloudCannon
- [ ] For "generic" components **only**: remove `- content_blocks` from `structures` in the YAML file (this was added for testing).

Commit:
Change checkbox and radio group alignmenrt

Commit:
Change gap back

Commit:
Don't render label if no label

Commit:
Add id to radio

Commit:
Add min-height to input

AlyshaNolan commented:
The alignment of the groups was supposed to stay the same - just the comment for the input needed changing

Commit:
change the alignment back

Commit:
Change the comments

Amber-Song commented:
> The alignment of the groups was supposed to stay the same - just the comment for the input needed changing

Ah. Yes. Changed the comment and changed the alignment back.

Commit:
Fix: added more to condition for empty label - worked for null but not for empty string

Commit:
Feat: added comment for radio button group default

Commit:
Feat: added comment for radio button group default

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

Commit:
Feat: provided comments for radio and checkbox groups recommending a label be added

Commit:
Merge branch 'fix/form-style' of github.com:CloudCannon/small-business-template-eleventy into fix/form-style

adon-cloudcannon reviewed (changes requested):
- Checkbox and radio group ‘alignment’ text needs changing - copied from labelled icon and not changed to this context ✅
- If no label is provided for a checkbox/radio group, the little space in the border should be removed (i.e. don’t render an empty element) ✅
- Comments should be added for form best practices - e.g. “You should provide a label for groups” or “You should have at least one radio option checked by default”… something like that? ✅
- Textarea input should have a min-height to stop user from shrinking it too small ✅
- **Bug: radio group - clicking labels doesn’t select the radio input** - this still appears to not work? Just adding a radio button group and adding two buttons, the IDs appear to be the same? Request another look please :)

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

Commit:
Test ratio button

Commit:
Remove hide group_name again

Commit:
Change ratio id name

Commit:
Using bookshop component

Commit:
Change id back

Commit:
Updated 1 file via CloudCannon.

Amber-Song commented:
> * **Bug: radio group - clicking labels doesn’t select the radio input** - this still appears to not work? Just adding a radio button group and adding two buttons, the IDs appear to be the same? Request another look please :)

Good spotting. It is fixed. This is because in ratio button group component radio buttons list, we didn't use bookshop component.

Commit:
Hide id again

Commit:
Merge branch 'fix/form-style' of github.com:CloudCannon/small-business-template-eleventy into fix/form-style

Commit:
Merge branch 'main' into fix/form-style

AlyshaNolan commented:
It probably shouldn't use the subcomponent but should pass through a UUID for each input

I think the subcomponent shouldn't be used because of the group_name and required inputs

AlyshaNolan commented on component-library/components/generic/form/radio-button-group/radio-button-group.bookshop.yml:
```diff
@@ -12,9 +12,7 @@ blueprint:
-  radio_buttons: 
-    - label:
-      checked: false
+  radio_buttons: [bookshop:generic/form/radio-input]
```
```suggestion
  radio_buttons:
    - label:
      checked:
      input_id:
```

Commit:
Use radio id instead of radio component

Commit:
Updated 1 file via CloudCannon.

Commit:
Try ratio id initiate

Commit:
Merge branch 'fix/form-style' of github.com:CloudCannon/small-business-template-eleventy into fix/form-style

Commit:
Updated 1 file via CloudCannon.

Commit:
Try nested ratio id

Commit:
Merge branch 'fix/form-style' of github.com:CloudCannon/small-business-template-eleventy into fix/form-style

Commit:
Add ratio_id uuid

Commit:
Updated 1 file via CloudCannon.

Commit:
Updated 1 file via CloudCannon.

Amber-Song commented:
@adon-cloudcannon @AlyshaNolan It works now.

AlyshaNolan commented:
There is a typo in the input (ratio_id -> radio_id) and then this looks great other than removing the content blocks from the page 😄

Commit:
Change typo and remove block

Commit:
Merge branch 'fix/form-style' of github.com:CloudCannon/small-business-template-eleventy into fix/form-style:

Commit:
Merge branch 'main' into fix/form-style

Commit:
Merge branch 'main' into fix/form-style
